src/Nethermind/Nethermind.JsonRpc.Test/RpcTest.cs

```diff
@@ -63,10 +60,26 @@ private static IJsonRpcService BuildRpcService<T>(T module) where T : class, IRp
         return service;
     }
 
-    public static JsonRpcRequest GetJsonRequest(string method, params string[]? parameters)
+    public static JsonRpcRequest GetJsonRequest(string method, params string?[]? parameters)
     {
-        var doc = JsonDocument.Parse(JsonSerializer.Serialize(parameters?.ToArray()));
-        var request = new JsonRpcRequest()
+        // NOTE: Since parameters are already passed as strings we want to avoid double serialization of JSON objects
+        var sb = new StringBuilder("[");
+        foreach (var param in parameters ?? [])
+        {
+            if (!string.IsNullOrEmpty(param) && param[0] == '{')
+            {
+                sb.Append(param);
+            }
+            else
+            {
+                sb.Append(JsonSerializer.Serialize(param));
+            }
+            sb.Append(',');
+        }
+        sb.Append(']');
```

Comment:
Alternatively maybe parameters, should be objects, that we would just set in `Params` property and they would get serialised?

Reply:
Solved by passing `object?[]?` as `parameters`.